dbms/src/Functions/FunctionsStringSearch.cpp

```diff
@@ -1063,6 +1063,166 @@ struct ReplaceStringImpl
         }
     }
 
+    // Handle the case where `column_src` and `replace` are const
+    static void vectorConstReplacement(
```

Comment:
Can we modify existing functions? Existing functions are already capable of handling a list of sources. It should be better to allow it handling constant source (i.e. one source), minimizing the changes.

Reply:
I think these functions are necessary because the first parameter of existing functions is `Column` by default, which will be parsed into `ColumnString` in `FunctionsStringSearch.cpp`, and our implementation needs to parse the parameter into `ColumnConst`. `ColumnConst` does not have `getChars` and `getOffsets` methods, so existing functions cannot be used.

Reply:
It doesn't make sense. Think about it. You can now calculate a bunch of FN(A, B), why cannot calculate a single row of FN(ConstA, ConstB)?

Reply:
Yes, I found a way to convert between the two types, so that now the existing function can be used directly, and the amount of code is greatly reduced.